internal/transform/transform.go

```diff
@@ -442,7 +442,7 @@ func resolveIdForMatch(match *ImportMatch, opts *TransformOptions) string {
 }
 
 func eachImportStatement(doc *astro.Node, cb func(stmt js_scanner.ImportStatement) bool) {
-	if doc.FirstChild.Type == astro.FrontmatterNode {
+	if doc.FirstChild.Type == astro.FrontmatterNode && doc.FirstChild.FirstChild != nil {
```

Comment:
Yep, this is the correct fix! Empty frontmatter is always generated, so we need to check if it has a child or not before proceeding.